components/upload/UploadList.tsx

```diff
@@ -64,6 +64,13 @@ export default class UploadList extends React.Component<UploadListProps, any> {
 
   // https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsDataURL
   previewFile = (file: File | Blob, callback: Function) => {
+    const { previewFile } = this.props;
+
+    if (previewFile) {
+      previewFile(file, callback);
+      return;
+    }
+
```

Comment:
We should remove `readAsDataURL` from here, move it into `previewFile` demo.

Reply:
This will be break change. Move it as `previewFile` default prop may better.

Reply:
Try this https://stackoverflow.com/a/25813769